### ISSUE TITLE
Best results in ModelNet40

### ISSUE THREAD

**Issue by zeal-github** [erikwijmans/Pointnet2_PyTorch#12]
Thanks for your job! Pytorch is more elegant for me.
I want to asks that what's your best result of classification trained on ModelNet40 using the default hyper-parameters? Or, what's the best accuracy when you tune the hyper-parameters appropriate? 
I'm training the model using your code and I will be appreciated if you post the best results.

**Comment by erikwijmans** [comment]
This repo matches the performance from the paper.  If I recall correctly, the only hyper-parameter you need to change is the number of points (to 10k) for that.

**Comment by zeal-github** [comment]
@erikwijmans Thanks for your reply! My training results do match the paper, but it is still 0.x% accuracy gap with the paper. And I find that, in your implementation, the architecture seems to be pruned.
'''
        self.SA_modules.append(
            PointnetSAModuleMSG(
                npoint=512,
                radii=[0.1, 0.2, 0.4],
                nsamples=[32, 64, 128],
                `mlps=[[input_channels, 64], [input_channels, 128],` 
################### in the paper, it seem to be a three layer mlp for per sub-pointnet
                      [input_channels, 128]],
                use_xyz=use_xyz
            )
        )
'''
I want to know if you do this on purpose?
Thanks!

**Comment by erikwijmans** [comment]
I have played around with the architectures a fair amount.  Makes sense to change them back to the ones given in Charles' repo, I will make that change.

**Comment by erikwijmans** [comment]
The default parameters should do that. train/train_cls.py will train an MSG model on modelnet40.

**Comment by LiuNull** [comment]
> @erikwijmans Thanks for your reply! My training results do match the paper, but it is still 0.x% accuracy gap with the paper. And I find that, in your implementation, the architecture seems to be pruned.
> 
> ```
>         self.SA_modules.append(
>             PointnetSAModuleMSG(
>                 npoint=512,
>                 radii=[0.1, 0.2, 0.4],
>                 nsamples=[32, 64, 128],
>                 `mlps=[[input_channels, 64], [input_channels, 128],` 
> ################### in the paper, it seem to be a three layer mlp for per sub-pointnet
>                       [input_channels, 128]],
>                 use_xyz=use_xyz
>             )
>         )
> ```
> I want to know if you do this on purpose?
> Thanks!

Hi, sorry to bother. Did you match the paper's accuracy with the newest arch? I run the code and only got 0.9023 with the default hyper-parameters.
